### DISCUSSION TITLE
Does NuxtImage with ipx cache the generated images on the server when using SSR?

### DISCUSSION THREAD

**Discussion by JenBerry** [nuxt/nuxt#25462] [Questions]
I'm a bit confused about how NuxtImage with ipx works, and I'm struggling to find answers in the documentation (https://image.nuxt.com/providers/ipx). How can we get an optimised image without needing an external service like cloudinary? Does ipx generate each image again every time it is requested, or does it/can it store them somewhere? I'm using a google cloud run instance which doesn't have it's own persistent storage so I would need a hook somewhere send them to storage. Can it be used in conjunction with a CDN?

Thanks

**Comment by bloor** [comment]
Basically NUXT spins up an IPX server.
Then when you access a photo, it redirects it to this server which:
A) if image was never processed OR the cache expired ; generates a new instance
B) if image exists and valid; it serves it

### **The issue is, the default cache is set to 60s.** ~_~ .. I lost 1 day to understand this.
So what you need is a config where you specify a `maxAge` for the cached image on the IPX server.

```js
    image: {
        format: ['avif'],
        provider: 'ipx',
        ipx: {
            modifiers: {
                quality: 90
            },
            maxAge: 86400  //24h
            
        }
    }, 
```

Now with this config, on production, the processed images are cached/stored for 24h. 
The first visit of the image (by any user) generates the result , which is returned for every other access by any other user. So technically only 1 user will notice the "image processing" loading time when viewing the image every 24h.
